src/engine/ruleset/decoders/sonicwall/firewall-generated.yml

```diff
@@ -0,0 +1,165 @@
+    # id=undeo sn=loremip time="2016-8-16 8:45:59" fw=10.134.0.141 pri=very-high c=uis m=1149 msg="idolore" n=onse fw_action="cancel"
+    - event.original: >-
+        <~log.fill> time="<@timestamp/%Y-%m-%d %H:%M:%S>" <~log.fill> m=<event.code> msg="<~log.msg_action>" <~log.fill> fw_action="<~log.fw_action>"
+
+normalize:
+  - map:
+      - event.kind: event
+      - event.module: sonicwall
+      - event.dataset: sonicwall.firewall
+      - fileset.name: firewall
+      - input.type: log
+      - observer.product: Firewalls
+      - observer.type: Firewall
+      - observer.vendor: Sonicwall
+      - rsa.internal.messageid: $event.code
+      - rsa.time.event_time: $@timestamp
+      - rsa.network.dinterface: $observer.egress.interface.name
+      - rsa.network.sinterface: $observer.ingress.interface.name
+      - rsa.db.index: $~rsa.db.index
+      - tags: +a_append/forwarded/sonicwall.firewall
+      - rsa.identity.user_sid_dst: $~rsa.identity.user_sid_dst
+  - check:
+      - host.name: +ef_exists
+      - source.address: +ef_exists
+    map:
+      - related.hosts: +a_append/$host.name/$source.address
+  - check:
+      - destination.address: +ef_exists
+    map:
+      - related.hosts: +a_append/$destination.address
+      - rsa.network.host_dst: $destination.address
+  - check:
+      - source.ip: +ef_exists
+      - destination.ip: +ef_exists
+    map:
+      - related.ip: +a_append/$source.ip/$destination.ip
+  - check:
+      - source.nat.ip: +ef_exists
+      - destination.nat.ip: +ef_exists
+    map:
+      - related.ip: +a_append/$source.nat.ip/$destination.nat.ip
+  - check:
+      - ~log.start_message: +ef_not_exists
+    logpar:
+      - rsa.time.event_time: <~log.year>-<~log.mounth>-<~log.day>T<~log.fill>
+    map:
+      - aux_slash: '/'
+      - rsa.time.date: +s_concat/$~log.year/$aux_slash/$~log.mounth/$aux_slash/$~log.day
+      - aux_slash: +ef_delete
+  - check:
+      - ~log.note: +ef_exists
+    logpar:
+      - ~log.note: <~log.note.header> <~log.note.protocol.header>:<~log.note.protocol.payload>
+    map:
+      - network.protocol: $~log.note.protocol.payload
+  - check:
+      - ~log.fw_action: +ef_exists
+    map:
+      - event.action: $~log.fw_action
+      - rsa.internal.event_desc: $~log.msg_action
+  - check:
+      - ~log.fw_action: +ef_not_exists
+      - ~log.msg_action: accept
+    map:
+      - event.action: $~log.msg_action
+  - check:
+      - ~log.fw_action: +ef_not_exists
+      - ~log.msg_action: allow
+    map:
+      - event.action: $~log.msg_action
+  - check:
+      - ~log.fw_action: +ef_not_exists
+      - ~log.msg_action: block
+    map:
+      - event.action: $~log.msg_action
+  - check:
+      - ~log.fw_action: +ef_not_exists
+      - ~log.msg_action: cancel
+    map:
+      - event.action: $~log.msg_action
+  - check:
+      - ~log.fw_action: +ef_not_exists
+      - ~log.msg_action: deny
+    map:
+      - event.action: $~log.msg_action
+  - check: >-
```

Comment:
I do not understand this, why do we map event.action to block, deny, or cancel  if we are going to delete it here anyways?

Reply:
There are log events that do not always bring a cancel, deny, allow, accept or block. In those cases the expected json does not bring the event.action field. These values were extracted from the elastic pipeline.json